test_utils/basetest.py

```diff
@@ -46,38 +20,62 @@ def resource_group_exists(self, group_name):
-        resource_group_params = {'location': self.location}
-        resp = self.resource_client.resource_groups.delete(
-            self.RESOURCE_GROUP_NAME, resource_group_params)
-        resp.wait()
-        print('Deleting {}'.format(self.RESOURCE_GROUP_NAME), resp.status())
+        try:
+            resp = self.resource_client.resource_groups.begin_delete(self.RESOURCE_GROUP_NAME)
+            resp.wait()
+            print('Deleting {}'.format(self.RESOURCE_GROUP_NAME), resp.status())
+        except Exception as e:
+            print("An unexpected error occurred during the delete_resource_group:")
+            print("Exception", e)
 
     def create_resource_group(self):
-        resource_group_params = {'location': self.location}
-
-        resp = self.resource_client.resource_groups.create_or_update(
-            self.RESOURCE_GROUP_NAME, resource_group_params)
-        print('Creating {}'.format(
-            self.RESOURCE_GROUP_NAME),
-            resp.properties.provisioning_state)
+        try:
+            resource_group_params = {'location': self.namespace_location}
+            resp = self.resource_client.resource_groups.create_or_update(self.RESOURCE_GROUP_NAME, resource_group_params)
+            print('Creating {}'.format(self.RESOURCE_GROUP_NAME), resp.properties.provisioning_state)
+        except Exception as e:
+            print("An unexpected error occurred during the create_resource_group:")
+            print("Exception", e)
 
     def deploy_template(self):
-        print("Deploying template")
-        deployment_name = "%s-Test-%s" % (datetime.datetime.now().strftime(
-            "%d-%m-%y-%H-%M-%S"), self.RESOURCE_GROUP_NAME)
-        template_data = self._parse_template()
-        deployment_properties = {
-            'mode': DeploymentMode.incremental,
-            'template': template_data
-        }
-
-        deployresp = self.resource_client.deployments.create_or_update(
-            self.RESOURCE_GROUP_NAME,
-            deployment_name,
-            deployment_properties
-        )
-        deployresp.wait()
-        print("ARM Template deployed", deployresp.status())
+        try:
+            deployment_name = "%s-Test-%s" % (datetime.datetime.now().strftime("%d-%m-%y-%H-%M-%S"), self.RESOURCE_GROUP_NAME)
+            
+            template_file_path = os.path.join(os.path.abspath('..'), 'src', self.template_name)
+            parameters = {
+                'SumoEndpointURL': {'defaultValue': self.sumo_endpoint_url},
+                'sourceCodeBranch': {'defaultValue': self.branch_name},
+                'sourceCodeRepositoryURL': {'defaultValue': self.repo_name}
+            }
+
+            # Read the template file
+            with open(template_file_path, 'r') as template_file:
+                template_content = template_file.read()
+
+            # Define the deployment properties
+            deployment_properties = {
+                'mode': 'Incremental',
```

Comment:
do not hard code it, use the enum DeploymentMode.incremental

Reply:
updated